src/compactor_state.rs

```diff
@@ -106,11 +108,142 @@ impl CompactorState {
                 return Err(SlateDBError::InvalidCompaction);
```

Comment:
Would the check in the block `the compaction overwrites an existing sr but doesn't include the sr` still be required? Wouldn't the more comprehensive check you are implementing cover it?